### ISSUE TITLE
Consider using an explicit return value and function overloads for `centerCrop` (and maybe other utils)

### ISSUE THREAD

**Issue by corymharper** [DominicTobias/react-image-crop#507]
Currently, `centerCrop` is implicitly typed to return either `PixelCrop | PercentCrop`, even though if you pass it a percent crop object as the first argument it will always return a `PercentCrop` and vice versa. It seems like relatively small modifications would need to be made in order for the Typescript compiler to know exactly which type of crop object the function would return. This could probably be applied to other utils, particularly `makeAspectCrop` which is used in tandem with `centerCrop` in the documentation. Here's basically what I think could be done (haven't tested it yet):

```ts
function makeAspectCrop(crop: Pick<PercentCrop, 'unit'> & Partial<Omit<PercentCrop, 'unit'>>, aspect: number, containerWidth: number, containerHeight: number): PercentCrop;
function makeAspectCrop(crop: Pick<PixelCrop, 'unit'> & Partial<Omit<PixelCrop, 'unit'>>, aspect: number, containerWidth: number, containerHeight: number): PixelCrop;
export function makeAspectCrop(crop: Partial<Crop>, aspect: number, containerWidth: number, containerHeight: number): PixelCrop | PercentCrop {
```

**Comment by corymharper** [comment]
The use case for this is when you type state in the following manner:

```ts
const [crop, setCrop] = useState<PercentCrop>();
```

The above state setter will complain if you try to hand it the return value of `centerCrop` currently.

**Comment by DominicTobias** [comment]
Yep seems like a good idea to me. I'm travelling atm so if you want it done quickly please raise a PR!

**Comment by corymharper** [comment]
If I find the bandwidth I'll put in a pull request if I have the time before you do, its no absolute rush for me right now.

**Comment by DominicTobias** [comment]
This seems to work, I'll publish the change soon

**Comment by DominicTobias** [comment]
Released in 10.0.7
